org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/editor/contentassist/antlr/ContentAssistContextFactory.java

```diff
@@ -419,7 +435,16 @@ public String getPrefix(INode prefixNode) {
 	public String getNodeTextUpToCompletionOffset(INode currentNode) {
 		int startOffset = currentNode.getOffset();
 		int length = completionOffset - startOffset;
-		String nodeText = ((ILeafNode) currentNode).getText();
+		final String nodeText;
+		ICompositeNode root = currentNode.getRootNode();
+		if (root instanceof RootNode) {
```

Comment:
PLease do not cast to concrete types. root.getText() will yield the same. Also root should never be something else than a RootNode

Reply:
Because 'getRoot' returns a 'ICompositeNode' I was not sure about that.